utils/ohos/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+    URL_HASH SHA256=f0397ba6e982c4450f27bf32a2a83292aba035b827a5623a14636ea583318c41
+  )
+  set(BOOST_ENABLE_CMAKE ON)
+  set(BOOST_RUNTIME_LINK static)
+  set(BUILD_TESTING OFF)
+  FetchContent_MakeAvailable(Boost)
+  add_library(Boost_boost INTERFACE)
+  add_library(Boost::boost ALIAS Boost_boost)
+  target_include_directories(Boost_boost INTERFACE ${boost_SOURCE_DIR})
+endif()
+
+include(Helper)
+
+# List of WasmEdge options
+option(WASMEDGE_BUILD_TESTS "Generate build targets for the wasmedge unit tests." OFF)
+option(WASMEDGE_BUILD_COVERAGE "Generate coverage report. Require WASMEDGE_BUILD_TESTS." OFF)
+option(WASMEDGE_BUILD_AOT_RUNTIME "Enable WasmEdge LLVM-based ahead of time compilation runtime." ON)
+option(WASMEDGE_BUILD_SHARED_LIB "Generate the WasmEdge shared library." ON)
+option(WASMEDGE_BUILD_STATIC_LIB "Generate the WasmEdge static library." OFF)
+option(WASMEDGE_BUILD_TOOLS "Generate wasmedge and wasmedgec tools." ON)
+option(WASMEDGE_FORCE_DISABLE_LTO "Forcibly disable link time optimization when linking even in Release/RelWithDeb build." OFF)
+set(WASMEDGE_BUILD_PACKAGE "DEB;RPM" CACHE STRING "Package generate types")
+set(CPACK_PROJECT_CONFIG_FILE ${CMAKE_CURRENT_SOURCE_DIR}/cmake/cpack_config.cmake)
+
+if(WASMEDGE_BUILD_COVERAGE)
+  include(CodeCoverage)
+  append_coverage_compiler_flags()
+endif()
+
+include(GNUInstallDirs)
+
+string(TOUPPER "${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}" CMAKE_INSTALL_DEFAULT_COMPONENT_NAME_UPCASE)
+set(CPACK_PACKAGE_VENDOR Second State LLC)
+set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${CPACK_PACKAGE_VERSION}-${CMAKE_SYSTEM_NAME}")
+set(CPACK_STRIP_FILES ON)
+set(CPACK_PACKAGE_CONTACT "Shen-Ta Hsieh <beststeve@secondstate.io>")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "High performance WebAssembly Virtual Machine")
+set(CPACK_ARCHIVE_COMPONENT_INSTALL ON)
+set(CPACK_ARCHIVE_${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME_UPCASE}_FILE_NAME "${CPACK_PACKAGE_FILE_NAME}")
+set(CPACK_RPM_COMPONENT_INSTALL ON)
+set(CPACK_RPM_MAIN_COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME})
+set(CPACK_RPM_PACKAGE_LICENSE "Apache 2.0")
+set(CPACK_DEB_COMPONENT_INSTALL ON)
+set(CPACK_DEBIAN_${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME_UPCASE}_PACKAGE_NAME wasmedge)
+set(CPACK_DEBIAN_FILE_NAME DEB-DEFAULT)
+set(CPACK_DEBIAN_COMPRESSION_TYPE xz)
+set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "https://github.com/WasmEdge/WasmEdge/")
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+set(CPACK_DEBIAN_PACKAGE_CONTROL_STRICT_PERMISSION ON)
+set(CPACK_GENERATOR "${WASMEDGE_BUILD_PACKAGE}")
+set(CPACK_PACKAGE_DESCRIPTION "WasmEdge is a high performance, extensible, and hardware optimized WebAssembly Virtual Machine for cloud, AI, and blockchain applications.")
+
+if(WASMEDGE_BUILD_TESTS)
+  include(CTest)
+  add_subdirectory(test)
+endif()
+
+add_subdirectory(include)
+add_subdirectory(lib)
+add_subdirectory(thirdparty)
+if(WASMEDGE_BUILD_TOOLS)
+  add_subdirectory(tools)
+endif()
+
+include(CPack)
+include(CPackComponent)
```

Comment:
In fact, I don't know the way to import the origin one in parent dir. So I copy it and when user runs the build_for_ohos.sh, this one will replace the origin one.

Reply:
Maybe this can help. https://stackoverflow.com/questions/17228677/how-to-include-an-additional-cmakelists-txt

Because copying is a very dangerous way, it will be unsynced with the latest version from the master.